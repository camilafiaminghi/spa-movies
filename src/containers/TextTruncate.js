import React, { Fragment, Component } from 'react'
import PropTypes from 'prop-types'
import truncate from './../utils/truncate'
import './TextTruncate.scss'

class TextTruncate extends Component {
	static propTypes = {
		text: PropTypes.string.isRequired,
		len: PropTypes.number.isRequired
	}

	state = {
		show: false
	}

	handleToggle = () => {
		const { show } = this.state;
		this.setState((oldState) => ({...oldState, show: !show}));
	}

	render() {
		const { text } = this.props;
		const { show } = this.state;
		const paragraph = (!show) ? truncate(...Object.values(this.props)) : text;

		return (
			<Fragment>
				{paragraph}
				<button
					type="button"
					className="text-truncate-toggle"
					data-testid="plus"
					onClick={() => this.handleToggle()}>{(!show) ? ` show more`: ` show less`}</button>
			</Fragment>
		)
	}
}

export default TextTruncate
