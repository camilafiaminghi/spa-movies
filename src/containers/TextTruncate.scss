@import "./../scss/variables.scss";
@import "./../scss/mixins.scss";
@import "./../../node_modules/rfs/scss";

.text-truncate-toggle {
	color: $color-default;
	font-style: italic;
	text-decoration: underline;
	@include responsive-font-size(1em);

	&:hover {
		color: $color-default;
	}
}
