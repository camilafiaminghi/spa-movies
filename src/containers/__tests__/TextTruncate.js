import React from 'react'
import {
  render,
  fireEvent,
  cleanup
} from '@testing-library/react'
import TextTruncate from './../TextTruncate'

const props = {
	text: 'Ut lectus enim, gravida vestibulum imperdiet ut, efficitur nec erat. Proin ullamcorper auctor erat, sit amet laoreet risus euismod at. Proin vitae mi eget felis consectetur feugiat. Curabitur ornare, nisi et viverra rhoncus, felis mauris fermentum lacus, rutrum pharetra diam enim ut nisl. Integer pellentesque ex eu lorem porta rutrum. Nulla facilisi. Curabitur elementum bibendum ex, ut placerat nisi consequat sit amet. Mauris et ipsum lobortis, dignissim sapien id, tincidunt lorem. Nulla id quam orci. Aliquam et metus elit. Aenean aliquam posuere semper.',
	len: 20
}

describe('<TextTruncate />', () => {

	afterEach(cleanup)

	it('renders the component', () => {
		const container = render(<TextTruncate {...props} />)
  	expect(container.firstChild).toMatchSnapshot()
  })

  it(`limit text with max ${props.len} chars`, () => {
		const { container } = render(<TextTruncate {...props} />)
		expect(container.firstChild.length).toBeLessThan(props.len)
  })

  it('handle toggle show/hide whole text', () => {
		const { container, getByTestId } = render(<TextTruncate {...props} />)
		const button = getByTestId('plus')

		fireEvent.click(button)
		expect(container.firstChild.length).toBe(props.text.length)

		fireEvent.click(button)
		expect(container.firstChild.length).toBeLessThan(props.len)
  })
})
