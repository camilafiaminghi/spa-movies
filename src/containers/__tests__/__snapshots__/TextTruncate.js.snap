// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextTruncate /> renders the component 1`] = `undefined`;
